src/mod_muc/mod_muc_log.erl: Prevent XSS in MUC logs by
linkifying only a few known protocols (EJAB-850)

Debian bug #520852
CVE ID: CVE-2009-0934
Upstream bug: https://support.process-one.net/browse/EJAB-850
 
Fixed in upstream SVN revision 1879, amended in revision 2020.
 
===================================================================
--- a/src/mod_muc/mod_muc_log.erl	(revision 1878)
+++ b/src/mod_muc/mod_muc_log.erl	(revision 1879)
@@ -752,7 +752,8 @@
     S2 = element(2, regexp:gsub(S1, "\\&", "\\&amp;")),
     S3 = element(2, regexp:gsub(S2, "<", "\\&lt;")),
     S4 = element(2, regexp:gsub(S3, ">", "\\&gt;")),
-    S5 = element(2, regexp:gsub(S4, "[-+.a-zA-Z0-9]+://[^] )\'\"}]+", link_regexp(NoFollow))),
+    S5 = element(2, regexp:gsub(S4, "((http|https|ftp)://|(mailto|xmpp):)[^] )\'\"}]+",
+				link_regexp(NoFollow))),
     %% Remove 'right-to-left override' unicode character 0x202e
     element(2, regexp:gsub(S5, [226,128,174], "[RLO]")).
